packages/@aws-cdk/assertions/lib/assertions.ts

```diff
@@ -53,23 +55,50 @@ export class TemplateAssertions {
   /**
    * Assert that a resource of the given type and properties exists in the
    * CloudFormation template.
+   * By default, performs partial matching on the `Properties` key of the resource, via the
+   * `Match.objectLike()`. To configure different behavour, use other matchers in the `Match` class.
    * @param type the resource type; ex: `AWS::S3::Bucket`
    * @param props the 'Properties' section of the resource as should be expected in the template.
    */
   public hasResourceProperties(type: string, props: any): void {
-    const assertion = assert.haveResource(type, props, assert.ResourcePart.Properties);
-    assertion.assertOrThrow(this.inspector);
+    this.hasResource(type, Match.objectLike({
+      Properties: props instanceof IMatcher ? props : Match.objectLike(props),
```

Comment:
should this be `IMatcher.isMatcher(props)`?

Reply:
Can't put a function on an interface tho

Reply:
Yes, and also can't do an `instanceof`. For this reason, I cheated and made it an abstract class.